CMakeLists.txt

```diff
@@ -7,9 +6,22 @@ ENDIF()
 
 IF(NOT KOKKOS_HAS_TRILINOS)
   CMAKE_MINIMUM_REQUIRED(VERSION 2.8.11 FATAL_ERROR)
+  # hack in openmp
+  IF (Kokkos_ENABLE_OpenMP)
+    FIND_PACKAGE (OpenMP REQUIRED)
+    IF (OPENMP_FOUND)
+      SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+      SET (CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${OpenMP_C_FLAGS}")
+    ENDIF ()
+  ENDIF (Kokkos_ENABLE_OpenMP)
+  IF (Kokkos_ENABLE_CUDA)
+    SET (CMAKE_CXX_COMPILER "${CMAKE_SOURCE_DIR}/bin/nvcc_wrapper")
+  ENDIF ()
   INCLUDE(cmake/tribits.cmake)
 ENDIF()
 
+SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```

Comment:
Agreed. I have used something like this before: 
```
# check for C++11
INCLUDE (CheckCXXCompilerFlag)
CHECK_CXX_COMPILER_FLAG ("-std=c++11" COMPILER_SUPPORTS_CXX11)
CHECK_CXX_COMPILER_FLAG ("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
IF (COMPILER_SUPPORTS_CXX11)
  SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
ELSEIF (COMPILER_SUPPORTS_CXX0X)
  SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
ELSE()
  MESSAGE (STATUS "The compiler ${CMAKE_CXX_COMPILER} has no C++11 \
  support. Please use a different C++ compiler.")
ENDIF()
```
But the check failed with the intel compilers I was using (despite the fact that they did in fact support c++11), so I had reverted to this. The correct answer sadly is probably to make that check if it is using an intel compiler and do something (though I'm not sure what that something is).